src/main/java/org/twittertip/CoSignedIdentifierAttestation.java

```diff
@@ -0,0 +1,81 @@
+package org.twittertip;
+
+import org.bouncycastle.asn1.ASN1EncodableVector;
+import org.bouncycastle.asn1.ASN1Primitive;
+import org.bouncycastle.asn1.DERBitString;
+import org.bouncycastle.asn1.DERSequence;
+import org.bouncycastle.crypto.AsymmetricCipherKeyPair;
+import org.bouncycastle.crypto.params.AsymmetricKeyParameter;
+import org.tokenscript.attestation.SignedIdentifierAttestation;
+import org.tokenscript.attestation.core.ASNEncodable;
+import org.tokenscript.attestation.core.SignatureUtility;
+import org.tokenscript.attestation.core.Validateable;
+import org.tokenscript.attestation.core.Verifiable;
+
+public class CoSignedIdentifierAttestation implements ASNEncodable, Verifiable, Validateable {
+    private final SignedIdentifierAttestation att;
+    private final byte[] signature;
+    private final AsymmetricKeyParameter attestationVerificationKey;
+
+    public CoSignedIdentifierAttestation(SignedIdentifierAttestation att, AsymmetricCipherKeyPair subjectSigningKey) {
+        this.att = att;
+        this.signature = SignatureUtility.signPersonalMsgWithEthereum(att.getDerEncoding(), subjectSigningKey.getPrivate());
+        this.attestationVerificationKey = subjectSigningKey.getPublic();
+        if (!verify()) {
+            throw new IllegalArgumentException("The signature is not valid");
+        }
+    }
+
+    /**
+     * Constructor used for when we supply the signature separately
+     */
+    public CoSignedIdentifierAttestation(SignedIdentifierAttestation att, AsymmetricKeyParameter subjectPublicKey, byte[] signature) {
+        this.att = att;
+        this.signature = signature;
+        this.attestationVerificationKey = subjectPublicKey;
+        if (!verify()) {
+            throw new IllegalArgumentException("The signature is not valid");
+        }
+    }
+
+    public SignedIdentifierAttestation getUnsignedAttestation() {
+        return att;
```

Comment:
I think this method signature is a bit misleading, since the attestation returned is actually signed